metabase/metabase
# fix implicit date parsing from numbers

Closes https://github.com/metabase/metabase/issues/19118

## Changes

When we render Line/Area/Bar/Scatter charts we try to infer x-axis data type from numbers/strings when x-axis type is not manually specified by user. The function `dimensionIsTimeseries` from `frontend/src/metabase/visualizations/lib/timeseries.js` is used to detect if x-axis values contain implicit date time by trying to parse the value using moment `moment(rows[0] && rows[0][i], moment.ISO_8601).isValid()`. The important detail is that we use `moment.ISO_8601` format here. However, we did not specify the format later when we parse actual x-axis values to render them.

## How to verify

- New -> Native query
```
select 2020 as fiscalyear, 200 as Quantity union
select 2021 as fiscalyear, 100 as Quantity union
select 2022 as fiscalyear, 150 as Quantity
```

- Visualize as Line
- Set x-axis `fiscalyear`
- Set y-axis `quantity`
- Ensure X-ticks show correct values
